src/pkg/p2p/preheat/job.go

```diff
@@ -185,9 +185,15 @@ func (j *Job) Run(ctx job.Context, params job.Parameters) error {
 
 			myLogger.Infof("Check preheat progress: %s", s)
 
-			// Finished
-			if s.Status == provider.PreheatingStatusSuccess {
+			switch s.Status {
+			case provider.PreheatingStatusFail:
+				// Fail
+				return preheatJobRunningError(errors.Errorf("preheat failed: %s", s))
```

Comment:
errors.Errorf("preheat failed: %s", s),  s is a struct? not a string, so maybe need change to  %v or s.Status.

Reply:
s has `String()` method. So it's ok.

```go
func (ps *PreheatingStatus) String() string
```